gobblin-runtime/src/main/java/gobblin/runtime/LimiterFactory.java

```diff
@@ -0,0 +1,72 @@
+/* (c) 2014 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.runtime;
+
+import java.util.concurrent.TimeUnit;
+
+import com.google.common.base.Preconditions;
+
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.configuration.State;
+
+
+/**
+ * A factory class for {@link Limiter}s.
+ *
+ * @author ynli
+ */
+public class LimiterFactory {
+
+  /**
+   * Create a new {@link Limiter} instance.
+   *
+   * @param state a {@link State} instance carrying configuration properties
+   * @return a new {@link Limiter} instance
+   */
+  public static Limiter newLimiter(State state) {
```

Comment:
It seems this assumes a guarantee that a limiter is desired? I see that in TaskContext you are checking whether throttling is enabled before calling the factory, but in the future someone might not be checking that and assume LimiterFactory handles no-limiter graciously. Maybe check if limiter is enabled and return with no exception if it isn't? (either return null, or return a Default limiter with no limit, or an optional).